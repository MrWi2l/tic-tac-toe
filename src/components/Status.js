import React from 'react'
import calculateWinner from '../calculateWinner'

const StatusContainer = ({ squares, currentPlayer }) => {
  let status
  let state
  const winner = calculateWinner(squares)

  if (squares.filter(x => x !== 'X' && x !== 'O').length === 0) {
    status = 'DRAW'
    state = 'draw'
  } else if (winner) {
    status = `Winner is : ${winner}`
    state = 'winner'
  } else {
    status = `Next player : ${currentPlayer}`
  }
  return <div className={state}>{status}</div>
}

StatusContainer.defaultProps = {
  currentPlayer: '',
}

StatusContainer.propTypes = {
  squares: React.PropTypes.array.isRequired,
  currentPlayer: React.PropTypes.string.isRequired,
}

module.exports = StatusContainer
