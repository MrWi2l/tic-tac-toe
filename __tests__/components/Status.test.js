import React from 'react'
import { shallow } from 'enzyme'
import Status from '../../src/components/Status'

const currentPlayer = 'X'
const squares = Array(9).fill(null) // [null, null, null...]
const squaresDraw = [
  'X', 'O', 'X',
  'X', 'O', 'X',
  'O', 'X', 'O',
]
const squaresWinner = [
  'X', 'O', '',
  'O', 'X', '',
  '', '', 'X',
]

describe('<Status />', () => {
  it('Test EMPTY', () => {
    const wrapper = shallow(
      <Status
        squares={squares}
        currentPlayer={currentPlayer}
      />,
    )

    expect(wrapper.hasClass('')).toMatchSnapshot()
    expect(wrapper.text()).toMatchSnapshot()
  })

  it('Test DRAW [className + text]', () => {
    const wrapper = shallow(
      <Status
        squares={squaresDraw}
        currentPlayer={currentPlayer}
      />,
    )
    expect(wrapper.hasClass('draw')).toMatchSnapshot()
    expect(wrapper.text()).toMatchSnapshot()
  })

  it('Test Winner [className + text]', () => {
    const wrapper = shallow(
      <Status
        squares={squaresWinner}
        currentPlayer={currentPlayer}
      />,
    )
    expect(wrapper.hasClass('winner')).toMatchSnapshot()
    expect(wrapper.text()).toMatchSnapshot()
  })
})
