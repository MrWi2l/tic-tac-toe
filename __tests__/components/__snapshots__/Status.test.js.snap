exports[`<Status /> Test DRAW [className + text] 1`] = `true`;

exports[`<Status /> Test DRAW [className + text] 2`] = `"DRAW"`;

exports[`<Status /> Test EMPTY 1`] = `true`;

exports[`<Status /> Test EMPTY 2`] = `"Next player : X"`;

exports[`<Status /> Test Winner [className + text] 1`] = `true`;

exports[`<Status /> Test Winner [className + text] 2`] = `"Winner is : X"`;
